src/App.js

```diff
@@ -1,6 +1,36 @@
+import { useEffect, useState } from 'react';
+import axios from 'axios';
+import Post from './components/Post';
+
 function App() {
+  const [posts, setPosts] = useState([]);
+  const [nextPageNumber, setNextPageNumber] = useState(1);
+  const [hasNextPage, setHasNextPage] = useState(true);
+
+  const pageLink = `http://jsonplaceholder.typicode.com/posts?_page=${nextPageNumber}&_limit=20`;
+
+  const checkNextPage = links => {
+    links.includes('next') ? setNextPageNumber(nextPageNumber + 1) : setHasNextPage(false);
+  }
+
+  const requestPosts = () => {
+    axios.get(pageLink)
+      .then(response => {
+        setPosts([...posts, ...response.data])
+        checkNextPage(response.headers.link)
+      })
+      .catch(err => console.error(err));
+  }
```

Comment:
What do you think about not depending on React to manage when a new request should be triggered? The code we have right now depends on React to re-render the component to trigger the next request, but data fetching is not React's responsibility. Can we move the load of the posts to a function that is called only once in the component and only resolves the promise once all of them are loaded?